packages/cloud_firestore/cloud_firestore/lib/src/utils/codec_utility.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright 2017, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of cloud_firestore;
+
+class _CodecUtility {
```

Comment:
There's already a CodecUtility in the MethodChannel implementation in the platform_interface package. Do we need this here?

Reply:
Ah, this is to translate DocumentReferences... Maybe this could be avoided by having DocumentReference extend platform.DocumentReferencePlatform, similar to FieldValue, but it doesn't seem a blocker for this change!

Reply:
I don't see a copy of `CodecUtility` in the `MethodChannel` implementation. There's a `FirestoreMessageCodec ` but it's doing something different. So I think we're ok here.

However... I wouldn't recommend extending `DocumentReferencePlatform`. That class has fields with a different signature from `DocumentReference`, which are supposed to be hidden from the app-facing interface. We'd probably want to expose an empty superclass instead, like `FieldValue` did. I'm not sure what to call it, since `DocumentReferencePlatform` is already taken.